app/code/Magento/InventoryInStorePickupQuote/Model/ResourceModel/GetPickupLocationCodeByQuoteAddressId.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickupQuote\Model\ResourceModel;
+
+use Magento\Framework\App\ResourceConnection;
+
+/**
+ * Get Pickup Location identifier by quote address identifier.
+ */
+class GetPickupLocationCodeByQuoteAddressId
+{
+    private const ADDRESS_ID = 'address_id';
+    private const PICKUP_LOCATION_CODE = 'pickup_location_code';
+
+    /**
+     * @var ResourceConnection
+     */
+    private $connection;
+
+    /**
+     * @param ResourceConnection $connection
+     */
+    public function __construct(
+        ResourceConnection $connection
+    ) {
+        $this->connection = $connection;
+    }
+
+    /**
+     * Fetch pickup location identifier by quote address identifier.
+     *
+     * @param int $addressId
+     *
+     * @return string|null
+     */
+    public function execute(int $addressId): ?string
+    {
+        $connection = $this->connection->getConnection();
+        $table = $this->connection->getTableName('inventory_pickup_location_quote_address');
+
+        $select = $connection->select()
```

Comment:
Why don't you use Magento DB abstraction? I don't see anything special here to use directly Zend Select?

Reply:
I am not sure if I understand your comment. 
`$connection->select()` returns `\Magento\Framework\DB\Select`.

Reply:
Discussed: to skip.